bin/desi_daily_archive.sh

```diff
@@ -0,0 +1,119 @@
+    (
+    echo "${execName} [-B] [-h] [-j JOBS] [-s DIR] [-V] [-v]"
+    echo ""
+    echo "Generate checksum and backup jobs for spectro/redux/daily/tiles/archive."
+    echo ""
+    echo "    -B         = If set, do NOT submit batch jobs."
+    echo "    -h         = Print this message and exit."
+    echo "    -j JOBS    = Use JOBS directory to write batch files (default ${DESI_ROOT}/users/${USER}/jobs)."
+    echo "    -s DIR     = Use DIR for temporary files (default ${SCRATCH})."
+    echo "    -V         = Version. Print a version string and exit."
+    echo "    -v         = Verbose mode. Print extra information."
+    echo ""
+    ) >&2
+}
+#
+# Create jobs.
+#
+function create_archivedate_job() {
+    local tileid=$1
+    local archivedate=$2
+    local logs=$3
+    local job_name="redux_daily_tiles_archive_${tileid}_${archivedate}"
+    cat > ${jobs}/${job_name}.sh <<EOT
+#!/bin/bash
+#SBATCH --account=desi
+#SBATCH --qos=xfer
+#SBATCH --constraint=cron
+#SBATCH --time=30:00
+#SBATCH --job-name=${job_name}
+#SBATCH --output=${jobs}/%x-%j.log
+#SBATCH --licenses=cfs,scratch
+source /global/common/software/desi/desi_environment.sh main
+module load desida desiBackup
+source \${DESIDA}/bin/desida_library.sh
+set -o xtrace
+shopt -s extglob
+if [[ "${logs}" == "True" ]]; then
+    cd \${DESI_SPECTRO_REDUX}/daily/tiles/archive/${tileid}/${archivedate}/logs
+    sha256sum * > ${scratch}/${job_name}_logs.sha256sum
+    unlock_and_move ${scratch}/${job_name}_logs.sha256sum
+fi
+cd \${DESI_SPECTRO_REDUX}/daily/tiles/archive/${tileid}/${archivedate}
+sha256sum !(logs) > ${scratch}/${job_name}.sha256sum
+unlock_and_move ${scratch}/${job_name}.sha256sum
+cd \${DESI_SPECTRO_REDUX}/daily/tiles/archive
+htar -cvf desi/spectro/redux/daily/tiles/archive/${job_name}.tar -H crc:verify=all ${tileid}/${archivedate}
+[[ \$? == 0 ]] && mv -v ${jobs}/${job_name}.sh ${jobs}/done
+EOT
+    chmod +x ${jobs}/${job_name}.sh
+}
+#
+# Get options.
+#
+batch=true
+jobs=${DESI_ROOT}/users/${USER}/jobs
+scratch=${SCRATCH}
+verbose=false
+while getopts Bhj:s:Vv argname; do
+    case ${argname} in
+        B) batch=false ;;
+        h) usage; exit 0 ;;
+        j) jobs=${OPTARG} ;;
+        s) scratch=${OPTARG} ;;
+        V) version; exit 0 ;;
+        v) verbose=true ;;
+        *) usage; exit 1 ;;
+    esac
+done
+#
+# Create job directories.
+#
+[[ -d ${jobs}/done ]] || mkdir -p ${jobs}/done
+#
+# Find TILEID/ARCHIVEDATE directories.
+#
+batch_jobs_created='False'
+archive_dir=${DESI_SPECTRO_REDUX}/daily/tiles/archive
+for tile_dir in ${archive_dir}/*; do
+    tileid=$(basename ${tile_dir})
+    for archivedate_dir in ${tile_dir}/*; do
+        archivedate=$(basename ${archivedate_dir})
+        checksum_file=redux_daily_tiles_archive_${tileid}_${archivedate}.sha256sum
+        if [[ -f ${archivedate_dir}/${checksum_file} ]]; then
+            ${verbose} && echo "DEBUG: ${archivedate_dir}/${checksum_file} detected, skipping ${tileid}/${archivedate}."
```

Comment:
What happens if the checksum file was successfully created, but then the job crashed during the htar backup (HPSS unavailable, CFS I/O hang, unusual timeout, ...)?  It looks like that night would be skipped in future invocations. It looks like the jobs already has some logic for checking htar success and moving the script file to ${jobs}/done/ upon success.  How about checking for that instead of the checksum file?

Reply:
That's a fair point, but it also means we can never clean up the script files, or move them to another location (without rewriting the script). If you have ideas about that, please let me know. Otherwise I'll have to think about this some more.

Reply:
Also a fair point about the scripts. Perhaps this highlights that we shouldn't be using either the checksum file nor the script location for the dual purpose of X plus indicating that the job is done.  How about either:
* have the job create a dedicated sentinel file upon success; or
* have the wrapper script check HPSS directly for which tile-archivenights have been backed up

Reply:
The backups are tracked in the file `$CFS/desi/users/desi/jobs/redux_daily_tiles_archive.csv`, so now both the checksum file and that backup list are checked, and a batch file is created if either are missing.